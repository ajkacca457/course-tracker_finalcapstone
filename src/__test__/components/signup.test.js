import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import Signup from '../../components/signup';
import store from '../../redux/store';

describe('Home Component', () => {
  it('renders correctly', () => {
    const hometree = TestRenderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Signup />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(hometree).toMatchSnapshot();
  });

  test('renders the title text of the Login page', () => {
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
      </Provider>,
    );
    const title = getByText('Sign up for free!');
    expect(title).toBeInTheDocument();
  });

  test('renders button of Signup page', () => {
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
      </Provider>,
    );
    const button = getByText('Login');
    expect(button).toBeInTheDocument();
  });

  test('renders button of Signup page', () => {
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
      </Provider>,
    );
    const button = getByText('Sign up');
    expect(button).toBeInTheDocument();
  });
});
