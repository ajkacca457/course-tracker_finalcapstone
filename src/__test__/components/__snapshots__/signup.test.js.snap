// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<div
  className="logincont"
>
  <div
    className=" bg-white mx-auto p-2 rounded signupitem"
  >
    <h4
      className="text-center bg-info py-2 text-white mb-3 rounded"
    >
      Sign up for free!
    </h4>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group text-left"
      >
        <input
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Enter your name"
          type="text"
        />
      </div>
      <div
        className="form-group text-left"
      >
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Enter your username"
          type="text"
        />
      </div>
      <div
        className="form-group text-left"
      >
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
        />
      </div>
      <div
        className="form-group text-left"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        className="btn btn-info w-100"
        type="submit"
      >
        Sign up
      </button>
    </form>
    <div
      className="signupinfo text-right"
    >
      <p
        className="tagline"
      >
        Already have an account?
      </p>
      <a
        className="nav-link m-0 bg-danger w-25 text-center text-white rounded ml-auto tagbtn"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </div>
</div>
`;
